testsuite/cluster-test/src/experiments/cpu_flamegraph.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::experiments::ExperimentParam;
+use crate::{
+    aws, cluster::Cluster, experiments::Context, experiments::Experiment, instance,
+    instance::Instance,
+};
+
+use crate::effects::{Action, GenerateCpuFlamegraph};
+use anyhow::{format_err, Result};
+use async_trait::async_trait;
+use chrono::{Datelike, Timelike, Utc};
+use futures::future::FutureExt;
+use futures::join;
+use std::{
+    collections::HashSet,
+    fmt::{Display, Error, Formatter},
+    time::Duration,
+};
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+pub struct CpuFlamegraphParams {
+    #[structopt(
+        long,
+        default_value = "60",
+        help = "Number of seconds for which perf should be run"
+    )]
+    pub duration_secs: usize,
+}
+
+pub struct CpuFlamegraph {
+    duration_secs: usize,
+    perf_instance: Instance,
+}
+
+impl ExperimentParam for CpuFlamegraphParams {
+    type E = CpuFlamegraph;
+    fn build(self, cluster: &Cluster) -> Self::E {
+        let (perf_instance, _) = cluster.split_n_validators_random(1);
```

Comment:
Did not we agree to have some `random_validator` method?

Reply:
Sorry, I missed this. Updated.